client/signup/main.jsx

```diff
@@ -337,7 +339,11 @@ class Signup extends React.Component {
 	}
 
 	updateShouldShowLoadingScreen = ( progress = this.props.progress ) => {
-		const hasInvalidSteps = !! getFirstInvalidStep( this.props.flowName, progress );
+		const hasInvalidSteps = !! getFirstInvalidStep(
+			this.props.flowName,
+			progress,
+			this.props.isLoggedIn
+		);
```

Comment:
I think this helper and a few other ones would benefit from the following refactor:
```js
getFirstInvalidStep( flowName, progress, isLoggedIn );
```
to
```js
getFirstInvalidStep( flows.getFlow( flowName, isLoggedIn ), progress );
```

This PR introduces a `isLoggedIn` into a lot of places, only to pass it down to `getFlow`. Making the `getFlow` call explicit would add more clarity IMO.

Reply:
Yeah, I agree it would improve things. However, it sounds like work for a follow-up PR to me 😉

Reply:
Would you mind if I push a commit or two here? I feel rather strongly about this 🙂

Reply:
Not at all 😉 Go for it!

Reply:
Well, I tried and got stuck in a rabbit hole. Caused mainly by the fact that the return value of `flows.getFlow( name )` doesn't include a `name` field. I'll need a standalone PR to solve that. #53542 is the first step.